### DISCUSSION TITLE
Speed up configs with many/nested blocks.

### DISCUSSION THREAD

**Discussion by revnelson** [payloadcms/payload#10264] [Feature Requests & Ideas]
Is it possible to make a similar performance increase as the [createClientConfig](https://github.com/payloadcms/payload/pull/9457) change for blocks configs?

In our config we have 35 'standard' blocks, and 9 'wrapper' blocks. Each of the wrapper blocks has a 'blocks' field that can have most or all of the 35 standard blocks. Enabling all these causes each navigation in the admin panel to take several seconds (sometimes nearly 20 seconds to navigate between collections). By simply disabling the wrapper blocks, navigation is much more snappy. Unfortunately the wrapper blocks are integral to how we use payload, and the inability to reasonably navigate with them enabled is blocking our ability to migrate from v2 to v3.

On v2, this many blocks made HMR and build times agonizingly slow (10 to 20 minutes on each save/build). This is a huge pain-point for development but doesn't affect our editors because navigation is fine once built. I was very excited because with v3 iterative development is *much* faster, but it's never going to fly if our designers and editors have to wait 20 seconds for each navigation in the admin panel.

**Comment by AlessioGr** [comment]
We'll look into that!

If you're open to contributing, it would be incredibly helpful if you could provide a minimal project that reproduces these performance issues with the least amount of complexity. Ideally, this could be submitted as a commit to our [_community test suite](https://github.com/payloadcms/payload/tree/main/test/_community). This will allow us address this problem faster

**Reply by revnelson** [reply]
Thanks Alessio! I have created a stress-test with one of my more complex blocks generated 50 times, with a container block including those 50 blocks also generated 50 times. While this is considerably more than I'm currently using in the real-world, it's conceivable that a very large project could hit these numbers.

https://github.com/revnelson/payload-blocks

I'm happy to do a pull request for the _community test suite but am unclear about the best way to go about it with the unified package. Should I fork the entire payload package and insert the blocks/fields into that _community config and create a pull request?
